packages/repository-tests/src/crud/relations/acceptance/has-many-without-di.relation.acceptance.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Getter} from '@loopback/context';
+import {
+  DefaultCrudRepository,
+  Entity,
+  EntityCrudRepository,
+  hasMany,
+  HasManyRepositoryFactory,
+  juggler,
+  model,
+  property,
+} from '@loopback/repository';
+import {expect} from '@loopback/testlab';
+import {
+  deleteAllModelsInDefaultDataSource,
+  withCrudCtx,
+} from '../../../helpers.repository-tests';
+import {
+  CrudFeatures,
+  CrudRepositoryCtor,
+  CrudTestContext,
+  DataSourceOptions,
+} from '../../../types.repository-tests';
+
+export function hasManyWithoutDIRelationAcceptance(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: CrudRepositoryCtor,
+  features: CrudFeatures,
+) {
+  describe('HasMany relation without di (acceptance)', () => {
+    before(deleteAllModelsInDefaultDataSource);
+    // Given a Customer and Order models - see definitions at the bottom
+    let existingCustomerId: string;
+    let ds: juggler.DataSource;
+    let customerRepo: CustomerRepository;
+    let orderRepo: OrderRepository;
+
+    before(
+      withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+        ds = ctx.dataSource;
+        givenOrderRepository();
+        givenCustomerRepository();
+        await ctx.dataSource.automigrate([Customer.name, Order.name]);
+      }),
+    );
+
+    beforeEach(async () => {
+      await orderRepo.deleteAll();
+      existingCustomerId = (await givenPersistedCustomerInstance()).id;
+    });
+
+    it('can create an instance of the related model', async () => {
+      async function createCustomerOrders(
+        customerId: string,
+        orderData: Partial<Order>,
+      ): Promise<Order> {
+        return customerRepo.orders(customerId).create(orderData);
+      }
+      const order = await createCustomerOrders(existingCustomerId, {
+        description: 'order 1',
+      });
+      // avoid type problems of BSON type of mongodb
+      if (features.convertIdType) {
+        // eslint-disable-next-line require-atomic-updates
+        existingCustomerId = existingCustomerId.toString();
+      }
+      expect(order.toObject()).containDeep({
```

Comment:
```ts
// avoid type problems of BSON type of mongodb
if (features.convertIdType) {
```

I find this very problematic. Does it mean that relations are not working with MongoDB?

Have you tried the following assertion instead?

```ts
expect(toJSON(order)).to.containDeep(toJSON({
  customerId: existingCustomerId,
  description: 'order 1',
});
```

If relations are not working with MongoDB, then let's introduce a flag to skip all relation tests. I am concerned that the current flag `features.convertIdType ` is modifying tests to execute scenarios that are different from what LB4 users are going to do.

BTW I think the underlying problem is in `toObject` that's behaving the same way as `toJSON` and converting ObjectID values to strings. I believe that in LB3, `toObject` was implemented differently and preserved `ObjectID` values. I think this is worth fixing, but not as part of this pull request - I opened a new issue https://github.com/strongloop/loopback-next/issues/3607.

Reply:
> I find this very problematic. Does it mean that relations are not working with MongoDB?

To clarify: IIUC, `covertIdType` transformation is required to make relations work. IMO, applications are not going to apply this `convertIdType` transformation. Does that mean relations won't work for such apps?